src/components/MapView.tsx

```diff
@@ -819,15 +824,58 @@ class MapView extends NativeBridgeComponent(
     return this._runNativeCommand('showAttribution', this._nativeRef);
   }
 
+  _runNativeCommand<RefType, ReturnType = NativeArg>(
+    methodName: string,
+    nativeRef: RefType | undefined,
+    args?: NativeArg[],
+  ): Promise<ReturnType> {
+    if (NativeMapViewModule) {
+      // when this method is called after component mounts, the ref is not yet set
+      // schedule it to be called after a timeout
+      if (!this._nativeRef) {
+        return new Promise((resolve) => setTimeout(resolve, 1)).then(() => {
+          return this._runNativeCommand(methodName, this._nativeRef, args);
+        });
+      }
+
+      const handle = findNodeHandle(nativeRef as any);
+
+      // @ts-expect-error TS says that string cannot be used to index NativeMapViewModule.
+      // It can, it's just not pretty.
+      return NativeMapViewModule[methodName]?.(
+        handle,
+        ...(args ?? []),
+      ) as Promise<ReturnType>;
+    } else {
+      return super._runNativeCommand<RefType, ReturnType>(
+        methodName,
+        nativeRef,
+        args,
+      );
+    }
+  }
+
   _onPress(e: NativeSyntheticEvent<{ payload: GeoJSON.Feature }>) {
     if (isFunction(this.props.onPress)) {
-      this.props.onPress(e.nativeEvent.payload);
+      const { payload } = e.nativeEvent;
+
+      if (typeof payload === 'string') {
+        this.props.onPress(JSON.parse(payload));
+      } else {
+        this.props.onPress(payload);
+      }
     }
   }
 
   _onLongPress(e: NativeSyntheticEvent<{ payload: GeoJSON.Feature }>) {
     if (isFunction(this.props.onLongPress)) {
-      this.props.onLongPress(e.nativeEvent.payload);
+      const { payload } = e.nativeEvent;
+
+      if (typeof payload === 'string') {
+        this.props.onLongPress(JSON.parse(payload));
+      } else {
+        this.props.onLongPress(payload);
+      }
```

Comment:
This seems to be a repeating pattern so something might this might work better:
```suggestion
      this.props.onLongPress(this.decodePayload(e.nativeEvent.payload))
```

And I'd probably prefer checking for Fabric than checking the type of payload. What do you think?

Reply:
I've added a comment about it in the code, but I've done it this way because only on iOS new arch the payload is actually a string. Android on the new arch doesn't seem to utilize the new event emitters, so it doesn't enforce the strict type compact with the generated code.